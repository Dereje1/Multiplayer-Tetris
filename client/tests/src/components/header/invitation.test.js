import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Invitation from '../../../../src/components/header/invitation';

test('will render the invitation modal', () => {
    const wrapper = shallow(
        <Invitation
            onAcceptInvite={jest.fn()}
            onDeclineInvite={jest.fn()}
            invite={['stub_Invite_name', 3]}
        />)
    expect(toJson(wrapper)).toMatchSnapshot();
})

test('will empty render on no invite', () => {
    const wrapper = shallow(
        <Invitation
            onAcceptInvite={jest.fn()}
            onDeclineInvite={jest.fn()}
            invite={null}
        />)
    expect(wrapper.isEmptyRender()).toBe(true);
})

