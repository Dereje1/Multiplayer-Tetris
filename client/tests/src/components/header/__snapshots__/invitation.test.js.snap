// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render the invitation modal 1`] = `
<ForwardRef(Dialog)
  aria-labelledby="simple-dialog-title"
  open={true}
>
  <ForwardRef(DialogTitle)
    id="simple-dialog-title"
  >
    stub_Invite_name invites you to a duel!
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(Chip)
      avatar={
        <ForwardRef(Avatar)>
          3
        </ForwardRef(Avatar)>
      }
      color="primary"
      label="Difficulty"
      size="medium"
      style={
        Object {
          "marginLeft": 70,
        }
      }
    />
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(ButtonGroup)
      aria-label="vertical outlined primary button group"
      color="primary"
    >
      <ForwardRef(Button)
        onClick={[Function]}
        size="large"
        style={
          Object {
            "background": "green",
            "marginBottom": 20,
            "marginLeft": 10,
          }
        }
        variant="contained"
      >
        Accept
      </ForwardRef(Button)>
      <ForwardRef(Button)
        onClick={[Function]}
        size="large"
        style={
          Object {
            "background": "red",
            "marginBottom": 20,
            "marginLeft": 50,
          }
        }
        variant="contained"
      >
        Decline
      </ForwardRef(Button)>
    </ForwardRef(ButtonGroup)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
